CHANGELOG.md

```diff
@@ -10,10 +10,77 @@ Changes are grouped as follows
+
+#### Change: `retrieve_dataframe`
+- Previously, fetching was constricted to either raw- OR aggregate datapoints. This restriction has been lifted and the method now works exactly like the other retrieve-methods (with a few extra options relevant only for pandas `DataFrame`s).
+- Used to fetch time series given by `id` and `external_id` separately - this is no longer the case. This gives a significant, additional speedup when both are supplied.
+- The `complete` parameter has been removed and partially replaced by `uniform_index` (bool) which covers a subset of the previous features (with some modifications: now gives uniform index all the way from first given `start` to last given `end`). Rationale: Weird and unintuitive syntax (passing a string using a comma to separate options).
+- Interpolating, forward-filling or in general, imputation (also controlled via the `complete` parameter) is completely removed as the resampling logic *really* should be up to the user fetching the data to decide, not the SDK.
+- New parameter `column_names` to pick either `id`s or `external_id`s as the dataframe column names. Previously a mix of was used when both were supplied.
+Read more below in the removed section or check out the method's updated documentation.
+
+### Fixed
+- **Critical**: Fetching aggregate datapoints now works properly with the `limit` parameter. In the old implementation, `count` aggregates were first fetched to split the time domain efficiently - but this has little-to-no informational value when fetching *aggregates* with a granularity, as the datapoints distribution can take on "any shape or form". This often led to just a few returned batches of datapoints due to miscounting.
+- Fetching datapoints using `limit=0` now returns 0 datapoints, instead of "unlimited". This is now aligned with the API.
+- Removing aggregate names from columns in a Pandas `DataFrame` in the previous implementation, used `Datapoints._strip_aggregate_name`, but this removed everything after the first pipe character (`|`) it found, rendering it useless for all tags with such a character as part of their names. Now has to match the pipe character AND the exact aggregate name to be removed.
+- The method `Datapoints.to_pandas` could return `dtype=object` for numeric time series when all aggregate datapoints were missing; which is not *that* unlikely, e.g. when using `interpolation` aggregate on a `is_step=False` time series with datapoints spacing above 1 hour on average. In such cases, an object array only containing `None` would be returned instead of float array dtype with `NaN`s. Correct dtype is now enforced by an explicit cast.
```

Comment:
```suggestion
- The method `Datapoints.to_pandas` could return `dtype=object` for numeric time series when all aggregate datapoints were missing; which is not *that* unlikely, e.g., when using `interpolation` aggregate on a `is_step=False` time series with datapoints spacing above one hour on average. In such cases, an object array only containing `None` would be returned instead of float array dtype with `NaN`s. Correct dtype is now enforced by an explicit cast.
```